presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpcds;
+
+import com.facebook.presto.spi.RecordCursor;
+import com.facebook.presto.spi.RecordSet;
+import com.facebook.presto.spi.type.DecimalParseResult;
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.Decimals;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.teradata.tpcds.Results;
+import com.teradata.tpcds.column.Column;
+import com.teradata.tpcds.column.ColumnType;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.spi.type.Decimals.rescale;
+import static com.facebook.presto.tpcds.TpcdsMetadata.getPrestoType;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Double.parseDouble;
+import static java.lang.Integer.parseInt;
+import static java.lang.Long.parseLong;
+import static java.util.Objects.requireNonNull;
+
+public class TpcdsRecordSet
+        implements RecordSet
+{
+    private final Results results;
+    private final List<Column> columns;
+
+    private final List<Type> columnTypes;
+
+    public TpcdsRecordSet(Results results, List<Column> columns)
+    {
+        requireNonNull(results, "results is null");
+
+        this.results = results;
+        this.columns = ImmutableList.copyOf(columns);
+        this.columnTypes = ImmutableList.copyOf(this.columns.stream().map(column -> getPrestoType(column.getType())).collect(Collectors.toList()));
```

Comment:
`ImmutableCollectors` instead of `ImmutableList.copyOf(stream.collect(toList()))`